File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        count=0
        temp=""
        result=""
        for _ in s:
            if _=='(':
                count+=1
            else:
                count-=1
            temp+=_
            if count==0:
                result+=temp[1:-1]
                temp=""
        return result
        
                

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        StringBuilder sb = new StringBuilder();
        int count = 0;
        String ans = "";
        for(int i=0; i<s.length(); i++)
        {
            if(s.charAt(i) == '(') count++;
            else count--;
            
            sb.append(s.charAt(i));
            if(count == 0)
            {
                ans+=sb.substring(1, sb.length()-1);
                sb.delete(0, sb.length());
            }
        }
        return ans;
    }
}